pkg/httphelper/request_logger.go

```diff
@@ -0,0 +1,38 @@
+package httphelper
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/net/context"
+	log "github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+)
+
+func NewRequestLogger(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		rw, ok := w.(*ResponseWriter)
+		if !ok {
+			panic("httphelper: Expected *httphelper.ResponseWriter")
+		}
+
+		start := time.Now()
+		addr := req.Header.Get("X-Real-IP")
+		if addr == "" {
+			addr = req.Header.Get("X-Forwarded-For")
```

Comment:
This technically needs parsing, I think it is comma separated, look in the router for details.